website/src/routes/guides/(migration)/migrate-from-zod/index.mdx

```diff
@@ -1,7 +1,101 @@
 ---
 title: Migrate from Zod
+description: Im based on Zod, so migrating should not require much work, because the naming and behaviour of methods is very similar.
 ---
 
+import { Link } from '@builder.io/qwik-city';
+
 # Migrate from Zod
 
-> The content of this page is not yet ready. Check back in a few days.
+I am heavily based on Zod, you could say it was the inspiration for my own creation. That's why a migration should not require much work, because the naming and behaviour of methods is very similar.
+
+## Setup
+
+First you need to install me. See <Link href="/guides/installation/">Installation instructions</Link>.
+
+## Schema Migration
+
+After this setup you will have to change every **Zod Schema**,
+but you don't have to change all of them just yet.
+
+> **This is a Process, so take your time!** *You can always take a break, and I will still be here, waiting for you.*
+
+Theres usually two ways to write these Schemas with Zod:
+
+### The `z` Object
+
+> Zod makes heavy use of it's `z` object export, which contains every schema type and validation function. This is one of Zod's disadvantages, because modern bundlers can't remove unused functions. Learn more <Link href="/guides/comparison/">here</Link>.
+
+You need to import every Schema and <Link href="/guides/pipelines/">Pipeline</Link> method you are using. Then you can just remove all references of the `z` object, and many of the Schema functions should be compatible.
+
+### Separate imports
+
+If your already using the different Schemas separately, then you only need to additionally import some of the <Link href="/guides/pipelines/">Pipeline</Link> methods. Again, many of the Schemas are compatible.
+
+## Example
+
+Let me explain it to you with an example.
+
+```ts
+import { z } from 'zod'; // 13.1 kB
```

Comment:
Probably not only this place but I found this way of showing size is kind of confusing since this is gzipped size. If not gzipped, this should be around 52 kB. It's confusing since gzip is not the only option (e.g. brotli)

Maybe we should use raw size in every place in the docs

Reply:
I understand your argument. However, I think that specifying 52 kB will cause more confusion than the gzip value. As a compromise, we could add "minified + gziped" to the comments in the docs.

Reply:
> we could add "minified + gziped" to the comments in the docs.

Sure, I think this is reasonable 👍

Reply:
Im working on this File again and I want to improve this. I don't want to lie here, I want to be precise. Zod is 13.1 kB minified + gziped but im not sure if some of the classes will be treeshaken. What's the best way of calculating these numbers?

Reply:
It also depends on which bundler we use to bundle / minify (and gzip) and what to import.
I checked the following tools:

- https://pkg-size.dev/zod: "55KB minified (13KB gzip)"
- https://bundlephobia.com/package/zod@3.21.4 "57kB MINIFIED 13.1kB MINIFIED + GZIPPED"

To get more precise result, we can just build it with real source code by trusted production grade tools like Webpack.

I created [this StackBlitz](https://stackblitz.com/edit/stackblitz-starters-tvsaau?file=src%2Findex.js) and it shows gzipped one is **10.8 kB**. I used `import * as z from 'zod'` to get some tree-shaking.
Probably this is the most fair metrics for Zod. BTW in terms of notation, I think something like "10.8 kB gzip" is easy to understand ("minified" is redundant when we say "gzip" since there is no reason not to minify when doing gzip)

BTW I also checked [the size of Valibot using StckBlitz](https://stackblitz.com/edit/stackblitz-starters-yufn18?file=src%2Findex.js), which is **562 bytes gzipped**. Notice that using `import * as v from 'valibot'` is just fine